.changes/unreleased/Added-20240920-122701.yaml

```diff
@@ -0,0 +1,9 @@
+kind: Added
+body: |-
+  Apply `ignore` function parameter metadata on any Directory passed from the CLI.
+
+  This extends the usage of `ignore` to not only dir loaded with `defaultPath` but any directory passed as function's param.
```

Comment:
See https://github.com/dagger/dagger/pull/8436#discussion_r1768373644 - I think we should ship both of these together, it's a weird functionality gap to only do one of them (especially given it doesn't seem tricky to do them at the same time?)

But if we decide to ship them separately, then this sentence should be rephrased for clarity - it's not *any* directory passed as a function param, it's *only* for directories passed on the CLI.

Reply:
> (especially given it doesn't seem tricky to do them at the same time?)

Yes the function I did was quite simple but it broke all the tests for a weird reason where python had missing file, Helder suggested to disable the double ignore and it fixed the issue.

I'll update the release note